pokemon_pong.py

```diff
@@ -0,0 +1,168 @@
+            self.dy = -1*self.dy
+        elif self.ball.bottom >= window_height: #hits bottom and changes direction
+            self.dy = -1*self.dy
+        elif self.ball.right >= window_width: #resets ball to center if it hits the left or right sides
+            p1_score()
+            self.ball.center = (600,600)
+            self.dx = -1*self.dx
+        elif self.rect.left <= 0:
+            p2_score()
+            self.ball.center = (600,600)
+            self.dx = -1*self.dx
+
+class Pikachu2(Pikachu):
+    def __init__(self):
+        pygame.sprite.Sprite.__init__(self)
+        self.image = pygame.image.load('pikachu.png').convert_alpha()
+        self.image = pygame.transform.scale(self.image, (150,200)) 
+        self.rect = self.image.get_rect() 
+        self.pikac = self.rect
+        self.pikac.centery = window_height / 2.0 #setting starting location
+        self.pikac.right = window_width
+
+    def move_pika2(self,key):
+        if 150 < self.rect.bottom < window_height:
+            if key[pygame.K_DOWN]:
+                self.rect.centery += 25
+            elif key[pygame.K_UP]:
+                self.rect.centery -= 25
+
+        elif self.rect.bottom >= window_height:
+            self.rect.centery -= 25
+        elif self.rect.bottom <= 150:
+            self.rect.centery += 25
+
+'''Scoring'''
+def p1_score():
+    global p1_wins
+    p1_wins += 1
+    win_game()
+
+def p2_score():
+    global p2_wins
+    p2_wins += 1
+    win_game()
+
+def win_game(): #displays win message and resets score and player positions if someone reaches the score 5
+    global p1_wins, p2_wins
+    if p1_wins == 5: 
+        p1_wins = 0
+        p2_wins = 0
+        p1_wins_text = win_text.render('Player 1 wins!', 1, (255,255,255))
+        screen.blit(p1_wins_text, (300,600))
+        pygame.display.flip()
+        pygame.display.update()
+        time.sleep(2)
+        reset_game()
+    elif p2_wins == 5:
+        p1_wins = 0
+        p2_wins = 0
+        p2_wins_text = win_text.render('Player 2 wins!', 1, (255,255,255))
+        screen.blit(p2_wins_text, (300,600))
+        pygame.display.flip()
+        pygame.display.update()
+        time.sleep(2)
+        reset_game()
+
+def reset_game(): #resets pikachu locations when someone wins the game
+    global pika, pika2 
+    pika.pikac.centery = window_height / 2.0
+    pika2.pikac.centery = window_height / 2.0
+
+'''main'''
+pygame.init()
+screen = pygame.display.set_mode((window_width, window_height))
+pygame.display.set_caption('Pokemon Pong')
+pika = Pikachu()
+pika2 = Pikachu2()
+pongball = PongBall(pika, pika2)
+running = True
+
+while running == True:
+    pika.move_pika(pygame.key.get_pressed())
+    pika2.move_pika2(pygame.key.get_pressed())
+    pongball.set_ball()
+    pongball.collision_wall()
+    #makes the ball bounce in another direction if it collides with a pikachu
+    if pygame.sprite.collide_rect(pongball, pika) or pygame.sprite.collide_rect(pongball, pika2):
```

Comment:
It might be nice to check this in a function or something, akin to pongball.collision_wall(), but I understand just checking for collision rectangles here.